service/worker/migration/fx.go

```diff
@@ -78,18 +82,33 @@ func NewResult(params initParams) fxResult {
 	}
 }
 
-func (wc *replicationWorkerComponent) Register(worker sdkworker.Worker) {
+func (wc *replicationWorkerComponent) RegisterWorkflow(worker sdkworker.Worker) {
 	worker.RegisterWorkflowWithOptions(ForceReplicationWorkflow, workflow.RegisterOptions{Name: forceReplicationWorkflowName})
 	worker.RegisterWorkflowWithOptions(NamespaceHandoverWorkflow, workflow.RegisterOptions{Name: namespaceHandoverWorkflowName})
 	worker.RegisterWorkflow(ForceTaskQueueUserDataReplicationWorkflow)
-	worker.RegisterActivity(wc.activities())
+
+	worker.RegisterActivity(wc.activities().GetMetadata) // may be run locally
 }
 
-func (wc *replicationWorkerComponent) DedicatedWorkerOptions() *workercommon.DedicatedWorkerOptions {
+func (wc *replicationWorkerComponent) DedicatedWorkflowWorkerOptions() *workercommon.DedicatedWorkerOptions {
 	// use default worker
 	return nil
 }
 
+func (wc *replicationWorkerComponent) RegisterActivities(worker sdkworker.Worker) {
+	worker.RegisterActivity(wc.activities())
+}
+
+func (wc *replicationWorkerComponent) DedicatedActivityWorkerOptions() *workercommon.DedicatedWorkerOptions {
+	return &workercommon.DedicatedWorkerOptions{
+		TaskQueue: primitives.MigrationActivityTQ,
+		Options: sdkworker.Options{
+			BackgroundActivityContext: headers.SetCallerType(context.Background(), headers.CallerTypeBackground),
```

Comment:
should default to CallerTypePreemptable I think.

Reply:
Last time we talked about it I thought I remembered we said migration should get higher priority than other background processes like delete namespace. Maybe @yux0 could weigh in here?

Reply:
`Background` right now basically means active task processing and can not be shared with replication or delete namespace. 

I am open to the idea of having different type for replication and namespace deletion. or rename `Background` to something else.

Reply:
I believe we can use CallerTypePreemptable here.